### PR TITLE
Feature/deform mesh using nodes skining

### PR DESCRIPTION
**Pull request by Erick9Thor**
Manual QA please, thanks.

### CODE REVIEW

File: Game/assets/shaders/vertex.glsl [hunk 1]
```diff
@@ -14,6 +20,13 @@ layout(std140, row_major, binding = 0) uniform Camera
 
 uniform mat4 model;
 
+/*layout(std140, row_major) uniform Skining
+{
+    mat4 palette[MAX_BONES];
+};*/
```
Comment (Erick9Thor) [inline Game/assets/shaders/vertex.glsl:26]: I will change this when we have batching and SSBO.

File: Source/src/components/ComponentMesh.cpp [hunk 2]
```diff
@@ -100,16 +129,18 @@ void Hachiko::ComponentMesh::UpdateSkinPalette(float4x4* palette) const {
         for (unsigned i = 0; i < mesh->num_bones; ++i)
         {
             const ResourceMesh::Bone& bone = mesh->bones[i];
-            const GameObject* bone_node = node_cache[i];
+            /* const GameObject* bone_node = node_cache[i];
```
Comment (Erick9Thor) [inline Source/src/components/ComponentMesh.cpp:132]: This is another improve that apears on the diapo called Cache bones. @baransrc if you want i can dedicate time to implemnt i think it's easy and improve the performance.

**Review by Bernatmago — APPROVED** [review]

File: Source/src/importers/MeshImporter.cpp [hunk 3]
```diff
@@ -138,17 +152,42 @@ Hachiko::Resource* Hachiko::MeshImporter::Load(UID id)
...
+    {
+        mesh->src_bone_weights = std::make_unique<float4[]>(sizes[static_cast<int>(ResourceMesh::Buffers::BONES_WEIGHTS)]);
+        size_bytes = sizeof(float4) * sizes[static_cast<int>(ResourceMesh::Buffers::BONES_WEIGHTS)];
+        memcpy(mesh->src_bone_weights.get(), cursor, size_bytes);
+        cursor += size_bytes;
+    }
+    else
+    {
+        mesh->src_bone_weights = nullptr;
+    }
```
Comment (baransrc) [inline Source/src/importers/MeshImporter.cpp:177]: I believe you follow what we had previously here. This is not a comment for you, but rather a note to the future us, we can delete all this `if else` hell, and just assign `nullptr` or `0` to everything at start and just leave the `if`s.

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the awesome job as always, I requested a small change, after that we are good to go!!🔥

**Review by monikamnc — APPROVED** [review]
Tested and working on my machine, 
I had to delete Library because I was having super strange errors but then it works perfectly.
I agree with @baransrc comments.

File: Source/src/core/GameObject.cpp [hunk 4]
```diff
@@ -526,5 +526,14 @@ Hachiko::GameObject* Hachiko::GameObject::GetFirstChildWithName(
...
 
+    for (GameObject* child : children)
+    {
+        GameObject* found_child = child->GetFirstChildWithName(child_name);
+        if (found_child != nullptr)
+        {
+            return found_child;
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (armandojaga) [reply Source/src/core/GameObject.cpp:537]: I agree that the name can be misleading as to what the intent is, I'd make some other changes as well, call the method `FindDescendant` and return `nullptr ` if it doesn't find anything

**Review by armandojaga — CHANGES_REQUESTED** [review]

**Review by baransrc — APPROVED** [review]

File: Source/src/core/GameObject.cpp [hunk 5]
```diff
@@ -528,3 +528,25 @@ Hachiko::GameObject* Hachiko::GameObject::GetFirstChildWithName(
...
+    {
+        GameObject* found_child = child->FindDescendantWithName(child_name);
+        if (found_child != nullptr)
+        {
+            return found_child;
+        }
+    }
+
+    return nullptr;
+}
```
Comment (alvarosoppelsa) [inline Source/src/core/GameObject.cpp:552]: Is there any reason for iterating two times over the children? if not, the content could be in just one for loop, I think is better for performance and for reading the code

**Review by armandojaga — APPROVED** [review]

**Pull request merged**